oasst-shared/oasst_shared/utils.py

```diff
@@ -1,6 +1,20 @@
+import time
 from datetime import datetime, timezone
 
+from loguru import logger
+
 
 def utcnow() -> datetime:
     """Return the current utc date and time with tzinfo set to UTC."""
     return datetime.now(timezone.utc)
+
+
+def log_timing(func):
+    def wrapped(*args, **kwargs):
```

Comment:
just for prettiness, could use `functools.wraps`

Reply:
I made it now a param + non-param decorator .. with optional logging of the kwargs. regarding `functool.wraps` ..that makes sure that docstring and function name are preserved .. right?

Reply:
yes, it's just for clarity